src/component/Calculator.js

```diff
@@ -0,0 +1,64 @@
+import React, { useState } from "react";
+import { operators } from "../const";
+import DigitButtons from "./DigitButtons";
+import ModifierButtons from "./ModifierButtons";
+import OperationButtons from "./OperationButtons";
+import "../css/index.css";
+
+// key는 어떤 값으로 지정할까?
+// total은 클릭 영역이 아님
+// 기능 분리하기
+
+export default function Calculator() {
+  const [calc, setCalc] = useState("0");
+
+  const getResult = (e) => {
+    let replace_str = calc.replace(/X/gi, "*");
+    if (isNaN(eval(replace_str))) return setCalc("0");
+    if (eval(replace_str) == Infinity) return setCalc("오류");
+    setCalc(eval(replace_str).toFixed(0));
+  };
+
+  const setResult = (e) => {
+    const inputText = e.target.innerText;
+    const className = e.target.className;
+    if (
+      className != "digit" &&
+      className != "modifier" &&
+      className != "operation"
+    )
+      return;
+
+    if (inputText == "AC") return setCalc("0");
+
+    // 연산자만, 연산자 연속 입력 불가능
+    const lastChar = calc.slice(-1);
+    if (operators.includes(lastChar) && isNaN(inputText)) return;
+
+    if (calc == "0" || calc == "오류") {
+      if (e.target.className == "operation") return;
+      setCalc("");
```

Comment:
초기 값이 "0"인데 빈 문자열로 초기화 할 이유가 있을까요?

Reply:
```
setCalc((prev) => {
      return prev + inputText;
});
```
의도는 setResult() 함수 가장 하위에 있는 위 구문을 활용하려 했던 것이긴 한데...
early return 일관성에도 맞지 않고, 불필요한 state 변경을 두 번이나 하고 있네요 🫢

setResult() 함수 전체적으로 계산과 검증로직이 섞여있어 함께 수정해보겠습니다!

```suggestion
      setCalc(inputText);
      return;
```

Reply:
앱을 초기화한다면 항상 초기 값을 잘 기억해두시고 초기값과 일치하는 값으로 하시는게 좋답니다!